documentation/api/query/tutorial-pql.markdown

```diff
@@ -0,0 +1,403 @@
+
+The `"="` operator follows a specific structure:
+
+`<attribute to compare> = <value>`
+
+In this case, the attributes are "type" and "title", and the values are "User"
+and "nick".
+
+The `"and"` operator also has a well-defined structure:
+
+`<query clause> and <query clause>`
+
+The query clauses can be any legal query (including another `"and"`). At least
+one clause must be specified, and all the clauses must be true for the
+`"and"` clause to be true. An `"or"` operator is also available, which looks
+just like the `"and"` operator, except that, as you'd expect, it's true if
+*any* specified clause is true.
+
+If we execute this query, we get results that look something like this:
+
+    [{
+      "parameters" : {
+        "comment" : "Nick Lewis",
+        "uid" : "1115",
+        "shell" : "/bin/bash",
+        "managehome" : false,
+        "gid" : "allstaff",
+        "home" : "/home/nick",
+        "groups" : "developers",
+        "ensure" : "present"
+      },
+      "line" : 111,
+      "file" : "/etc/puppetlabs/code/environments/production/manifests/user.pp",
+      "exported" : false,
+      "tags" : [ "firewall", "default", "node", "nick", "role::base", "users", "virtual", "user", "account", "base", "role::firewall::office", "role", "role::firewall", "class", "account::user", "office", "virtual::users", "allstaff" ],
+      "title" : "nick",
+      "type" : "User",
+      "resource" : "0ae7e1230e4d540caa451d0ade2424f316bfbf39",
+      "certname" : "foo.example.com"
+    }]
+
+Our results are an array of "resources", where each resource is an object with
+a particular set of keys.
+
+* `parameters`: this field is itself an object, containing all the parameters and values of the resource
+* `line`: the line the resource was declared on
+* `file`: the file the resource was specified in
+* `exported`: true if the resource was exported by this node, or false otherwise
+* `tags`: all the tags on the resource
+* `title`: the resource title
+* `type`: the resource type
+* `resources`: this is an internal identifier for the resource used by PuppetDB
+* `certname`: the node that the resource came from
+
+There will be an entry in the list for every resource. A resource is specific
+to a single node, so if the resource is on 100 nodes, there will be 100 copies
+of the resource (each with at least a different certname field).
+
+### Excluding results
+
+We know this instance of the user "nick" is defined on line 111 of
+`/etc/puppetlabs/code/environments/production/manifests/user.pp`. What if
+we want to check whether or not we define the same resource somewhere else?
+After all, if we're repeating ourselves, something may be wrong! Fortunately,
+there's an operator to help us:
+
+    resources {
+      type = "User" and
+      title = "nick" and
+      !(file = "/etc/puppetlabs/code/environments/production/manifests/user.pp" and line = 111)
+    }
+
+The `!` operator wraps another clause, and returns results for which the
```

Comment:
Up above you quote and backtick the operators. Minor nitpick but could we align them so they're all in the same format?